PracticeGit/test.c

```diff
@@ -0,0 +1,102 @@
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+// test.c
+// Program on A3 branch.
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+#pragma warning(disable:4996)
+#include <stdio.h>
+#include <limits.h>
+#include <string.h>
+#include "conversion.h"
+
+// Defines color codes for terminal output
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define YELLOW  "\033[1;33m"
+#define BLUE    "\033[1;34m"
+#define RESET   "\033[0m"
+
+void ProcessAndPrintConversions (long long inputNumber, char* formattedBinary, char* hexString) {
+   if (inputNumber < INT_MIN || inputNumber > INT_MAX) printf ("Value out of range for a 32-bit signed integer. Try again.\n");
+   else {
+      int n = (int)inputNumber;
+      DecimalToBinary (n);
+      FormatBinary (formattedBinary);
+      DecimalToHexadecimal (hexString);
+      printf ("Binary Equivalent for %d: % s\n", n, formattedBinary);
+      printf ("Hexadecimal Equivalent for %d: % s\n", n, hexString);
+   }
+}
+
+void ExecuteConversions (int isTest) {
+   char formattedBinary[BITS / 8 * 9 + 1], hexString[HEX_LENGTH + 1];
+   if (isTest) {

```

Comment:
```suggestion
   if (!isTest) return;
```
https://en.wikipedia.org/wiki/Guard_(computer_science)